### DISCUSSION TITLE
How do I reduce / shrink instance's disk size?

### DISCUSSION THREAD

**Discussion by alexander-potemkin** [canonical/multipass#3744] [Q&A] [answered]
It seems like I was a way too generous for the virtual instances in terms of the disk size and it seems like multipass makes disk allocation at the time of creation - is there a way to reduce disk size, without recreating the instance?

I have plenty of `Disk usage:     3.7GiB out of 19.3GiB` and my physical machine's disk space is kind of running out :)

Any help would be much appreciated!

**Comment by giuliazanchi** [comment]
Hi @alexander-potemkin , you can edit the resources allocated to an instance easily in the [Multipass GUI](https://multipass.run/docs/multipass-gui-client). If you don't have that installed, you can use the [`multipass set`](https://multipass.run/docs/set-command) command. You can find some examples and long-form instructions in [How to modify an instance](https://multipass.run/docs/modify-an-instance).

**Comment by andrei-toterman** [comment, marked as answer]
Hey, @alexander-potemkin! Shrinking the disk size through Multipass is unfortunately not possible. We can increase it, but we can't decrease it without the risk of data loss, which might negatively affect the VM. Also, the VM does not necessarily occupy as much space as the size of the disk it is reporting. It says 19.3GB, but if you only ever used, for example 5GB of space in that VM, that's how much space it is going to take on your disk, as it increases with use.

**Reply by alexander-potemkin** [reply]
Hey @andrei-toterman and thank you! May I kindly ask you for the hint - how could I verify the actual disk use for the disks images?

And why can't the steps like the [following](https://unix.stackexchange.com/questions/656836/how-do-i-shrink-a-partition-right-after-shrinking-its-filesystem-with-resize2fs) works?

**Reply by andrei-toterman** [reply]
Sure, I can tell you where to look for the image of the VM, and see how big that file is, but I'll need to know your operating system :)
Steps like those can in fact actually work. Depending on the Multipass driver that you are using (`multipass get local.driver`), you might be able to manually decrease the disk size. Again, that comes with the risk of data loss, so you have to be sure that you are shrinking the partition without deleting important stuff, and always make a backup of the image. If you tell me the operating system and driver, I might be able to point you in the right direction.

**Reply by alexander-potemkin** [reply]
Thank you! I'm on Ubuntu (server) and the driver is `qemu` :)

**Reply by andrei-toterman** [reply]
To see the size a VM occupies, go to `/var/snap/multipass/common/data/multipassd/vault/instances/<instance-name>`. There is an `.img` file. The size of that file is the amount of space the VM takes on your disk.

For shrinking the disk, there are many different approaches, there are a few here for example https://serverfault.com/questions/432119/is-there-any-way-to-shrink-qcow2-image-without-converting-it-raw, and I can't tell you which one is the best without trying them out, but if you want to find other instructions I would just look up `how to shrink a qcow2 image`. I hope you'll be able to do it!

**Reply by alexander-potemkin** [reply]
Thank you so much, that helps!!
